demo/test.cc

```diff
@@ -39,3 +45,34 @@ TEST(Demo, OK) {
   EXPECT_EQ(200, static_cast<int>(response.code));
   EXPECT_EQ("OK\n", response.body);
 }
+
+TEST(Demo, NoPointsYet) {
+  Singleton<DemoServer>();
+  const auto response = HTTP(GET("localhost:2015/demo_id"));
+  EXPECT_EQ(200, static_cast<int>(response.code));
+  EXPECT_EQ("{\"state\":{\"points\":[]}}\n", response.body);
+}
+
+TEST(Demo, AddThreePoints) {
+  Singleton<DemoServer>();
+  // TODO(dkorolev): Support JSON parsing on the server side, like this:
+  // HTTP(POST("localhost:2015/demo_id", JSON(State::Point(+0.25, -0.25, true)), "application/json")).body);
+  // TODO(dkorolev): Should be able to HTTP(POST(...)) JSON-s directly, w/o cerealize::JSON(), right?
+  // TODO(dkorolev): How about POST-s without HTML body? Do we need those extra parameters?
```

Comment:
Err, now I feel a bit confused. Aren't those parameters URL-encoded by
default?

Thanks,
Dima

http://dimakorolev.com/ | http://www.linkedin.com/in/dimakorolev |
dmitry.korolev@gmail.com | +1 (312) 593 2783


On Mon, Feb 16, 2015 at 6:25 PM, Dima Korolev <dmitry.korolev@gmail.com>
wrote:

> True, forgot about HTTP FORM-s. Is there perhaps a flag to have them send
> the values in the URL or in JSON? :-)
>
> Thanks,
> Dima
>
> http://dimakorolev.com/ | http://www.linkedin.com/in/dimakorolev |
> dmitry.korolev@gmail.com | +1 (312) 593 2783
>
>
> On Mon, Feb 16, 2015 at 6:23 PM, John Babak <notifications@github.com>
> wrote:
>
>> In demo/test.cc
>> <https://github.com/KnowSheet/KnowSheet/pull/8#discussion_r24788677>:
>>
>> > @@ -39,3 +45,34 @@ TEST(Demo, OK) {
>> >    EXPECT_EQ(200, static_cast<int>(response.code));
>> >    EXPECT_EQ("OK\n", response.body);
>> >  }
>> > +
>> > +TEST(Demo, NoPointsYet) {
>> > +  Singleton<DemoServer>();
>> > +  const auto response = HTTP(GET("localhost:2015/demo_id"));
>> > +  EXPECT_EQ(200, static_cast<int>(response.code));
>> > +  EXPECT_EQ("{\"state\":{\"points\":[]}}\n", response.body);
>> > +}
>> > +
>> > +TEST(Demo, AddThreePoints) {
>> > +  Singleton<DemoServer>();
>> > +  // TODO(dkorolev): Support JSON parsing on the server side, like this:
>> > +  // HTTP(POST("localhost:2015/demo_id", JSON(State::Point(+0.25, -0.25, true)), "application/json")).body);
>> > +  // TODO(dkorolev): Should be able to HTTP(POST(...)) JSON-s directly, w/o cerealize::JSON(), right?
>> > +  // TODO(dkorolev): How about POST-s without HTML body? Do we need those extra parameters?
>>
>> querystring in the URL and JSON in HTTP body are just enough. - sure, if
>> we don't want to support handling requests from simple HTML <form>s.
>> They put the form field values as query-string in the request body for <form
>> type="post"> or as the query-string in the URL for <form type="get">.
>> For more complex requests we'll have to use JavaScript XMLHttpRequest to
>> issue them (it's certainly not difficult in any way, just to be sure).
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/KnowSheet/KnowSheet/pull/8/files#r24788677>.
>>
>
>